merkle_tree/src/namespaced_merkle_tree/proof.rs

```diff
@@ -108,8 +109,17 @@ where
                 return Ok(Err(()));
             }
             // Verify the boundary proof
-            if <InnerTree<E, H, T, N, ARITY>>::verify(root, boundary_proof.index(), boundary_proof)?
-                .is_err()
+            // Create a commitment from the root hash with proper height calculation
+            let proof_height = boundary_proof.proof.len().checked_sub(1).ok_or_else(|| {
+                MerkleTreeError::InconsistentStructureError("Empty boundary proof".to_string())
+            })?;
+            let commitment = MerkleTreeCommitment::new(*root, proof_height, 0); // size won't be checked
+            if <InnerTree<E, H, T, N, ARITY>>::verify(
```

Comment:
I don't quite understand this code but it seems to be in line with how things work now.

Reply:
So basically we used to accept root node, now we only accept commitment, so we construct a commitment on the fly with num_leaves left unfilled since we don't need it for verification. 

It looks slightly hacky, I agree. But I don't want to expose another  "_unchecked" or special api for now.